rsfbclient-diesel/src/fb/connection.rs

```diff
@@ -0,0 +1,170 @@
+//! The Firebird connection
+
+use super::backend::Fb;
+use super::query_builder::FbQueryBuilder;
+use super::transaction::FbTransactionManager;
+use super::value::FbRow;
+use diesel::connection::*;
+use diesel::deserialize::*;
+use diesel::expression::QueryMetadata;
+use diesel::query_builder::bind_collector::RawBytesBindCollector;
+use diesel::query_builder::*;
+use diesel::result::Error::DatabaseError;
+use diesel::result::Error::DeserializationError;
+use diesel::result::*;
+use rsfbclient::SimpleConnection as FbRawConnection;
+use rsfbclient::{Execute, Queryable, Row, SqlType};
+use std::cell::RefCell;
+
+pub struct FbConnection {
+    pub raw: RefCell<FbRawConnection>,
+    tr_manager: FbTransactionManager,
+}
+
+impl SimpleConnection for FbConnection {
+    fn batch_execute(&self, query: &str) -> QueryResult<()> {
+        self.raw
+            .borrow_mut()
+            .execute(query, ())
+            .map_err(|e| DatabaseError(DatabaseErrorKind::__Unknown, Box::new(e.to_string())))
+            .map(|_| ())
+    }
+}
+
+impl Connection for FbConnection {
+    type TransactionManager = FbTransactionManager;
+    type Backend = Fb;
+
+    fn establish(database_url: &str) -> ConnectionResult<Self> {
+        #[cfg(feature = "pure_rust")]
+        let mut raw_builder = rsfbclient::builder_pure_rust();
+        #[cfg(any(feature = "linking", feature = "dynamic_loading"))]
```

Comment:
```suggestion
        // TODO: generalize to other backends
```
can we remove this at least for now?

1. I think this is not a good way to do it (not modular, and if `--all-features` is given then it does something unexpected)
2. `FbRawConnection` is hardcoded to native anyways